sklearn/metrics/classification.py

```diff
@@ -528,6 +528,7 @@ def matthews_corrcoef(y_true, y_pred, sample_weight=None):
     y_pred = lb.transform(y_pred)
 
     C = confusion_matrix(y_true, y_pred, sample_weight=sample_weight)
+    C = C.astype(np.float64)    # Bug#9622
```

Comment:
the bug number is not needed here, as long as the test fails if this line is removed (which it does).

Reply:
mathematically the `astype` is unnecessary and, in fact, could be viewed as dangerous, because it increases a possibility of a round-off errors.
the bug reference explains why we _do_ need it (because python lacks transparent bignums).

Reply:
I think you mean numpy lacks transparent bignums. I'm pretty sure Python by default does them.

I don't mind this terribly. If we keep it, write "Issue #9622" rather than "Bug#9622".

FWIW, to pass the test I found it sufficient to just replace
```
t_sum = C.sum(axis=1, dtype=np.float64)
p_sum = C.sum(axis=0, dtype=np.float64)
```
but perhaps in other cases the trace is the problem??

Reply:
That seems a decent approach!